CanHazFunny/CanHazFunny.Tests/JesterTests.cs

```diff
@@ -1,10 +1,86 @@
+﻿using System;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
 
 namespace CanHazFunny.Tests
 {
     [TestClass]
     public class JesterTests
     {
-        
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void JesterNewInstance_GivenNullJokeService_ThrowArgumentNullException()
+        {
+            // Arrange
+            // Act
+            Jester jester = new Jester(null!, new JokePrinter());
+            // Assert
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void JesterNewInstance_GivenNullJokePrinter_ThrowArgumentNullException()
+        {
+            // Arrange
+            // Act
+            Jester jester = new Jester(new JokeService(), null!);
+            // Assert
+        }
+
+        [TestMethod]
+        public void Jester_ImplementIJokeServiceAndJokePrinter_WorkingJokeServiceAndJokePrinter()
+        {
+            // Arrange
+            IJokeService iJokeService = new JokeService();
+            IJokePrinter iJokePrinter = new JokePrinter();
```

Comment:
In general we don't include the "i" in the variable name
```suggestion
            IJokeService jokeService = new JokeService();
            IJokePrinter jokePrinter = new JokePrinter();
```